pkg/controlplane/apiserver/options/options.go

```diff
@@ -262,6 +269,23 @@ func (o *Options) Complete(alternateDNS []string, alternateIPs []net.IP) (Comple
 			if err != nil {
 				return CompletedOptions{}, fmt.Errorf("failed to build token generator: %w", err)
 			}
+		} else if completed.ServiceAccountSigningEndpoint != "" {
+			plugin, cache, err := plugin.New(completed.Authentication.ServiceAccounts.Issuers[0], completed.ServiceAccountSigningEndpoint)
+			if err != nil {
+				return CompletedOptions{}, fmt.Errorf("while setting up external-jwt-signer: %w", err)
+			}
+			metadata, err := plugin.GetServiceMetadata(context.Background())
```

Comment:
consider plumbing the actual context for the server lifetime here if possible so we don't have to use context.Background()

Reply:
Done